### PR TITLE
Implementing BulkGetSecret

### PR DESCRIPTION
**Pull request by cmendible**
# Description

since https://github.com/dapr/components-contrib/issues/309 was closed this PR adds the `BulkGetSecret` api to dapr.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2168

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x]  Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: dapr/proto/runtime/v1/dapr.proto [hunk 1]
```diff
@@ -216,6 +219,22 @@ message GetSecretResponse {
   map<string, string> data = 1;
 }
 
+// GetBulkSecretRequest is the message to get the secrets from secret store.
+message GetBulkSecretRequest {
+  // The name of secret store.
+  string store_name = 1;
+
+  // The metadata which will be sent to secret store components.
+  map<string,string> metadata = 2;
```
Comment (yaron2) [inline dapr/proto/runtime/v1/dapr.proto:228]: Shouldn't this Request envelope hold an array of keys? I don't see where the array of keys are specified.

Comment (cmendible) [reply dapr/proto/runtime/v1/dapr.proto:228]: The idea of the new API was to get all the secrets of a given store (https://github.com/dapr/components-contrib/pull/310). 

Many of my customers use a dedicated store per app and they are used to get all secrets at once. For instance those who are using the Key Vault configuration provider for .NET expect it to load all secrets and merge it with the apps configuration and that is something I'm looking forward to add to this PR:   https://github.com/dapr/dotnet-sdk/pull/465 as discussed here: https://github.com/dapr/dapr/issues/2168.

**Comment by yaron2** [comment]
@cmendible can you resolve the conflicts please?

**Comment by yaron2** [comment]
@cmendible can you please resolve the conflict? we can then proceed to merge.

**Review by yaron2 — APPROVED** [review]
lgtm

**Comment by cmendible** [comment]
> @cmendible can you please resolve the conflict? we can then proceed to merge.

@yaron2 Conflicts resolved but now codecov is failing....

**Comment by yaron2** [comment]
> > @cmendible can you please resolve the conflict? we can then proceed to merge.
> 
> @yaron2 Conflicts resolved but now codecov is failing....

Let me worry about codecov :)

Do we have a docs PR for this change? Updating the API reference and the Secrets section

**Review by yaron2 — APPROVED** [review]

**Pull request merged**